module/VuFind/src/VuFindTest/Feature/LiveDetectionTrait.php

```diff
@@ -23,37 +24,37 @@
  * @package  Tests
  * @author   Demian Katz <demian.katz@villanova.edu>
  * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
- * @link     https://vufind.org Main Page
+ * @link     https://vufind.org/wiki/development:testing:unit_tests Wiki
  */
-namespace VuFindTest\Unit;
-
-use VuFindTest\Container\MockContainer;
+namespace VuFindTest\Feature;
 
 /**
- * Base class for tests using a MockContainer.
+ * Mix-in for detecting whether a live test environment is currently running.
  *
  * @category VuFind
  * @package  Tests
  * @author   Demian Katz <demian.katz@villanova.edu>
  * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
- * @link     https://vufind.org Main Page
+ * @link     https://vufind.org/wiki/development:testing:unit_tests Wiki
  */
-abstract class MockContainerTest extends \PHPUnit\Framework\TestCase
+trait LiveDetectionTrait
 {
     /**
-     * Mock container
+     * Flag to allow other traits to test for the presence of this one (to enforce
+     * dependencies).
      *
-     * @var MockContainer
+     * @var bool
      */
-    protected $container;
+    public $hasLiveDetectionTrait = true;
 
     /**
-     * Standard setup method.
+     * Is this test running in a continuous integration context?
      *
-     * @return void
+     * @return bool
      */
-    public function setUp(): void
+    public function continuousIntegrationRunning()
     {
-        $this->container = new MockContainer($this);
+        // We'll assume that if the CI Solr PID is present, then CI is active:
+        return file_exists(__DIR__ . '/../../../../../local/vufindtest.pid');
```

Comment:
Would it be possible to base this determination on an environment variable or something else more explicit?

Reply:
This file gets created here as part of the test instance spin up: https://github.com/vufind-org/vufind/blob/dev/build.xml#L339 -- I can't remember all of the history here, and I'm not sure without looking if this is still really a Solr PID file, or if it used to be and now it's just a relic used to set a flag for the tests. I'm open to changing it to something better, but I'm not sure what the best approach would be; the important thing is that whatever we use as a marker, it has to be set by the startup action and removed by the shutdown action, because most of the tests will run correctly regardless of whether or not the test environment is active, but a few need to know the state. Maybe a good starting point would be to review whether this is really a pid file or not, and at least rename it to something more appropriate if the name is no longer appropriate. I'll take a closer look next week when I'm back from my mini-vacation.

Reply:
I had a little time to investigate, and it does appear that the real Solr PID file got replaced with a fake placeholder at some point in time. I've opened [https://github.com/vufind-org/vufind/pull/1853](this PR) to restore reality to the situation. Does that make sense to you, or do you think we should come up with a different approach?

Reply:
I'm resolving this conversation since the work was resolved by PR #1853 (see discussion there for more details, if interested).